dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/MavenDependencyMediation.java

```diff
@@ -16,30 +16,27 @@
 
 package com.google.cloud.tools.opensource.classpath;
 
+import com.google.cloud.tools.opensource.dependencies.Artifacts;
+import com.google.cloud.tools.opensource.dependencies.DependencyGraph;
+import com.google.cloud.tools.opensource.dependencies.DependencyPath;
 import java.io.File;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
 import org.eclipse.aether.artifact.Artifact;
-import com.google.cloud.tools.opensource.dependencies.Artifacts;
-import com.google.cloud.tools.opensource.dependencies.DependencyPath;
 
-/**
- * Retain only the first version of a groupId:artifactId encountered.
- */
-class MavenDependencyMediation {
-  
+/** Retain only the first version of a groupId:artifactId encountered. */
+class MavenDependencyMediation implements DependencyMediation {
+
   private List<Artifact> artifacts = new ArrayList<>();
 
-  /**
-   * Returns true iff dependency mediation selects this artifact.
-   */
+  /** Returns true iff dependency mediation selects this artifact. */
   // TODO might be a problem if there's a classifier
-  boolean selects(Artifact artifact) {
+  private boolean selects(Artifact artifact) {
     return artifacts.contains(artifact);
   }
 
-  void put(DependencyPath dependencyPath) {
+  private void put(DependencyPath dependencyPath) {
     Artifact artifact = dependencyPath.getLeaf();
```

Comment:
The method `put` and `selects` assumed how they are called when traversing dependency graph. Because the call order was not applicable to Gradle, I didn't choose them as `DependencyMediation`'s interface method.